src/file_pipe_log/pipe_builder.rs

```diff
@@ -172,62 +172,61 @@ impl<F: FileSystem> DualPipesBuilder<F> {
         machine_factory: &FA,
     ) -> Result<(M, M)> {
         let threads = self.cfg.recovery_threads;
-        let (append_concurrency, rewrite_concurrency) =
-            match (self.append_files.len(), self.rewrite_files.len()) {
-                (a, b) if a > 0 && b > 0 => {
-                    let a_threads = std::cmp::max(1, threads * a / (a + b));
-                    let b_threads = std::cmp::max(1, threads.saturating_sub(a_threads));
-                    (a_threads, b_threads)
-                }
-                _ => (threads, threads),
-            };
         let pool = rayon::ThreadPoolBuilder::new()
             .num_threads(threads)
             .build()
             .unwrap();
-        let (append, rewrite) = pool.join(
-            || self.recover_queue(LogQueue::Append, machine_factory, append_concurrency),
-            || self.recover_queue(LogQueue::Rewrite, machine_factory, rewrite_concurrency),
-        );
-
+        // As the `recover_queue` would update the `Version` of each log file in
+        // `apend_files` and `rewrite_files`, we just use the naive api `install`
+        // as the choice for parallelly recovering `LogQueue::Append` and
+        // `LogQueue::Rewrite`.
+        let append =
+            pool.install(|| self.recover_queue(LogQueue::Append, machine_factory, threads));
+        let rewrite =
+            pool.install(|| self.recover_queue(LogQueue::Rewrite, machine_factory, threads));
```

Comment:
Sure, I'll supplement it.
On the one hand, an initial reason why i replaced `pool.join(_, _)` with seperated `install` is that `join` would cause `double-reference error` as the implementation of `recover_queue` introduces`&mut self`. 
On the other hand, each installed task would be excuted on the TheadPool by utilizing each sub-threads in it with the param `threads`.  And in my opinion, it seems also effective, compared to the original `join` with  param `rewrite_accuracy` and `append_accuracy` respectively ?

Reply:
Yes, I'm not saying the performance would be worse. It's just different.

Reply:
There still exists bugs when introducing `Criterion` crate for generating the performance report on MacOS.
I'll try to generate the comparison result in a later commit.